types/src/write_set.rs

```diff
@@ -2,16 +2,44 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! For each transaction the VM executes, the VM will output a `WriteSet` that contains each access
-//! path it updates. For each access path, the VM can either give its new value or delete it.
+//! path it updates. For each access path, the VM can either give its new value or delete it. If
+//! aggregator is used, a delta update is used.
 
 use crate::state_store::state_key::StateKey;
 use anyhow::Result;
 use aptos_crypto_derive::{BCSCryptoHash, CryptoHasher};
 use serde::{Deserialize, Serialize};
 
+/// Specifies operation such as +, - to use with `WriteOp::Delta`.
+#[derive(Copy, Clone, Hash, PartialOrd, Ord, PartialEq, Eq, Serialize, Deserialize)]
+pub enum DeltaOperation {
+    Addition,
+    Subtraction,
+}
+
+impl std::fmt::Debug for DeltaOperation {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DeltaOperation::Addition => write!(f, "+"),
+            DeltaOperation::Subtraction => write!(f, "-"),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Hash, PartialOrd, Ord, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DeltaLimit(pub u128);
+
+impl std::fmt::Debug for DeltaLimit {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "result <= {}", self.0)
+    }
+}
+
 #[derive(Clone, Eq, Hash, PartialEq, Serialize, Deserialize)]
 pub enum WriteOp {
     Deletion,
+    #[serde(skip)]
+    Delta(DeltaOperation, DeltaLimit, u128),
```

Comment:
is there a reason why you don't carry u128 in the DeltaOperation? like
enum DeltaOperation {
  Addition(u128),
  Subtraction(u128),
}

Reply:
Initially, the idea was to have `delta(op, bytes)` so that any update in the future cam simply specify op and serialize value into bytes. So no particular reason of keeping u128 in `Delta`. If we move it to operation as you suggested, I think we should move the limit as well since it is a postcondition of `DeltaOperation`

Reply:
I think we should merge the u128, I don't have strong opinion on DeltaLimit since it at least has a name 😄

Reply:
Cool, let's merge u128 only then for now :)